### PR TITLE
LDEV-4648 support partially escaped sequence (refiled on 6.1)

### PR DESCRIPTION
**Pull request by hideo67**
Hi !

This is the LDEV-4648 patch refiled on the 6.1 branch.

https://luceeserver.atlassian.net/browse/LDEV-4648

### CODE REVIEW

**Comment by hideo67** [comment]
The 6.1 branch doesn't pass the test suite.  Is this recognized by the dev team?

```
[INFO]      [java]    [script] -> JUnitReport: /home/hideo-t/work/java/Lucee/test/reports/junit-test-results-6.1.0.76-SNAPSHOT.xml
[INFO]      [java]    [script] -> Service [ postgres ]   not available,          7 tests skipped
[INFO]      [java]    [script] -> Service [ sftp ]       not available,          1 tests skipped
[INFO]      [java]    [script] -> Service [ ftp ]        not available,          1 tests skipped
[INFO]      [java]    [script] -> Service [ memcached ]          not available,          3 tests skipped
[INFO]      [java]    [script] -> Service [ ldap ]       not available,          7 tests skipped
[INFO]      [java]    [script] -> Service [ imap ]       not available,          5 tests skipped
[INFO]      [java]    [script] -> Service [ mongoDb ]    not available,          15 tests skipped
[INFO]      [java]    [script] -> Service [ s3_backblaze ]       not available,          1 tests skipped
[INFO]      [java]    [script] -> Service [ redis ]      not available,          1 tests skipped
[INFO]      [java]    [script] -> Service [ oracle ]     not available,          14 tests skipped
[INFO]      [java]    [script] -> Service [ s3 ]         not available,          51 tests skipped
[INFO]      [java]    [script] -> Service [ MSsql ]      not available,          51 tests skipped
[INFO]      [java]    [script] -> Service [ pop ]        not available,          5 tests skipped
[INFO]      [java]    [script] -> Service [ MySQL ]      not available,          101 tests skipped
[INFO]      [java]    [script] -> Service [ s3_google ]          not available,          1 tests skipped
[INFO]      [java]    [script] -> Service [ s3_custom ]          not available,          1 tests skipped
[INFO]      [java]    [script] -> Service [ smtp ]       not available,          38 tests skipped
[INFO]      [java]    [script] Failed: test.general.AdminPages
[INFO]      [java]    [script]  Fetch and test admin pages
[INFO]      [java]    [script]  /admin/index.cfm?action=services.update returned status code: 500. Expected [200] Actual [500]
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/general/AdminPages.cfc:148
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/general/AdminPages.cfc:78
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/_testRunner.cfc:283
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/run-tests.cfm:284
[INFO]      [java]    [script]
[INFO]      [java]    [script] Failed: test.tickets.LDEV1107
[INFO]      [java]    [script]  checking PNGimageRotation3() for PNG format, with height 4032 and width 3024
[INFO]      [java]    [script]  Expected [3024] but received [4032]
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/tickets/LDEV1107.cfc:14
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/_testRunner.cfc:283
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/run-tests.cfm:284
[INFO]      [java]    [script]
[INFO]      [java]    [script] Failed: test.tickets.LDEV1107
[INFO]      [java]    [script]  checking ImageInfo() for JPG format, with height 4032 and width 3024
[INFO]      [java]    [script]  Expected [3024] but received [4032]
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/tickets/LDEV1107.cfc:27
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/_testRunner.cfc:283
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/run-tests.cfm:284
[INFO]      [java]    [script]
[INFO]      [java]    [script] Failed: test.tickets.LDEV1107
[INFO]      [java]    [script]  checking PNGimageRotation3(), when image rotate into 90 degree in clockwise direction
[INFO]      [java]    [script]  Expected [3024] but received [4032]
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/tickets/LDEV1107.cfc:39
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/_testRunner.cfc:283
[INFO]      [java]    [script]          /home/hideo-t/work/java/Lucee/test/run-tests.cfm:284
[
```

File: core/src/main/java/lucee/commons/net/URLDecoder.java [hunk 1]
```diff
@@ -42,58 +44,20 @@ public static String decode(String str, boolean force) {
 	}
 
 	public static String decode(String s, String enc, boolean force) throws UnsupportedEncodingException {
-		if (!force && !ReqRspUtil.needDecoding(s)) return s;
```
Comment (michaeloffner) [inline core/src/main/java/lucee/commons/net/URLDecoder.java:45]: That check exists for good reason and should not be removed

Comment (hideo67) [reply core/src/main/java/lucee/commons/net/URLDecoder.java:45]: Ok.  I will revert the line.

Comment (hideo67) [reply core/src/main/java/lucee/commons/net/URLDecoder.java:45]: rebased on the current 6.1 branch.

**Comment by hideo67** [comment]
rebased on the current 6.1 branch

**Pull request merged**